syntax = "proto3";

message TickerRequest {
    string ticker_symbol = 1;
}

message StockUpdate {
    string ticker_symbol = 1;
    double current_price = 2;
    double price_change = 3;
    double change_percent = 4;
    int64 timestamp = 5;
    int32 volume = 6;
}

service StockTicker {
    rpc SubscribeToStockUpdates(stream TickerRequest) returns (stream StockUpdate);
}